### PR TITLE
[make] split up specific make vars/targets into frags

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Related issue**: #496 

**Type of change**: new feature/other enhancement

**Impact**: other

**Release Notes**
This changes the make system to have project-specific changes be put into a special `mk` file instead of being in the Chipyard main `mk`s. This allows for sub-projects to inject requirements, sources, flags and more into the build system using the `EXTRA_*` variables. This also adds `dramsim` as a default flag to all simulations. Finally, this also changes Verilators LDFLAGS to use `.a` libraries instead of the long thing that Rocket Chip uses. 

To see how this affects things with a more in-depth example see #496

### CODE REVIEW

File: common.mk [hunk 1]
```diff
@@ -3,6 +3,18 @@
 #########################################################################################
 SHELL=/bin/bash
 
+#########################################################################################
+# extra make variables/rules from subprojects
+#
+# EXTRA_GENERATOR_REQS - requirements needed for the main generator
```
Comment (jerryz123) [inline common.mk]: This is currently unused, I guess this is for dromajo?

Comment (abejgonzalez) [reply common.mk]: `GENERATOR_REQS` is for Ariane

File: generators/tracegen/tracegen.mk [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+##############################################################
+
+AXE_DIR=$(base_dir)/tools/axe/src
+AXE=$(AXE_DIR)/axe
+
+$(AXE): $(wildcard $(AXE_DIR)/*.[ch]) $(AXE_DIR)/make.sh
+	cd $(AXE_DIR) && ./make.sh
+
+$(output_dir)/tracegen.out: $(sim)
+	mkdir -p $(output_dir) && $(sim) $(PERMISSIVE_ON) +dramsim +max-cycles=$(timeout_cycles) $(SIM_FLAGS) $(EXTRA_SIM_FLAGS) $(VERBOSE_FLAGS) $(PERMISSIVE_OFF) none </dev/null 2> $@
```
Comment (jerryz123) [inline generators/tracegen/tracegen.mk]: Why does this still use +dramsim and +max-cycles?

Comment (abejgonzalez) [reply generators/tracegen/tracegen.mk]: I missed this. I only find/replaced in the `common.mk` and `Makefiles`

**Comment by jerryz123** [comment]
I like this change a lot, but I think someone else should sign-off on this

**Pull request merged**